social_media/accounts/views.py

```diff
@@ -0,0 +1,104 @@
+from django.contrib.auth import authenticate
+from django.shortcuts import render
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework.authtoken.models import Token
+from .serializer import AccountRegistrationSerializer, AccountSerializer
+from .models import User
+
+
+# Create your views here.
+class UserDetailsView(APIView):
+
+    def get(self, request, username):
+        if User.objects.filter(username=username).exists():
+            user = User.objects.get(username=username)
+            account_serializer = AccountSerializer(instance=user)
+            return Response(account_serializer.data)
+        else:
+            return Response(data={'error': {'message': 'user not found'}}, status=status.HTTP_404_NOT_FOUND)
+
+
+class LoginView(APIView):
+    def post(self, request):
+        data = request.data
+        email = data.get('email', None)
+        password = data.get("password", None)
+        if not email or not password:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+
+        # check if a user with such username or email exist
+        user = None
```

Comment:
You can use .filter() here instead since you ultimately need the object. 

.filter() -> when you need the object
.exists() -> when you just need to check if it exists() or not 

Also why have you assigned user=None? Where is user=None being used?